examples/contracts/signal_judge.py

```
@@ -0,0 +1,177 @@
+
+        # prompt_comparative: validators agree on the boolean correct field;
+        # current_price is allowed to differ by cents across validator fetches
+        raw = gl.eq_principle.prompt_comparative(
+            get_judgment,
+            "The boolean field 'correct' must have the same value across all answers. Ignore any differences in 'current_price' (which varies by cents because it is fetched at different timestamps) and ignore differences in 'reasoning_quality' (which is a subjective rating).",
+        )
+
+        try:
+            judgment = json.loads(raw)
+        except json.JSONDecodeError as e:
+            raise gl.vm.UserError(f"LLM did not return valid JSON: {e}")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)